webapp/lib/platform.dart

```diff
@@ -42,6 +42,33 @@ init() async {
     _docStorage = FirebaseDocStorage(firebase.firestore());
     _pubsubInstance = new PubSubClient(platform_constants.publishUrl, user);
     controller.command(controller.UIAction.userSignedIn, new controller.UserData(user.displayName, user.email, photoURL));
+
+    // Get user features
```

Comment:
This code is difficult for me to read - as this callback is actually executed below the call below - could we swap the order?

I wonder if we could also refactor it into a support method that that takes a data object and applies to cull the duplication a bit?

Reply:
Follow on PR is fine :)

Reply:
Thanks @lukechurch  I agree it was a bit unwieldy, and an accidental mis-indent was also making it harder to read.

I've refactored setting the flags into a separate function, and while doing so it occurred to me that it would be a good idea to always use the default settings from Firebase over the initial UI-set settings rather than only when there are no specific user preferences, in order to support single flag enables per user, if that makes sense?

Anyway, please have another look at this part of the code (L45-75) and let me know if you have any other concerns. Thanks!